trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/ConfigurationManagerService.java

```diff
@@ -225,7 +225,7 @@ public Response putConfiguration(@Parameter(description = CFG_TYPE_ID) @PathPara
             @RequestBody(description = CFG_UPDATE_DESC + " " + CFG_KEYS_DESC, content = {
                     @Content(examples = @ExampleObject("{\"parameters\":{" + CFG_PATH_EX +
                             "}}"), schema = @Schema(implementation = org.eclipse.tracecompass.incubator.internal.trace.server.jersey.rest.core.model.ConfigurationQueryParameters.class))
-            }, required = true) ConfigurationQueryParameters queryParameters) {
+            }, required = true) InputConfigurationQueryParameters queryParameters) {
```

Comment:
Maybe there is an issue with Swagger generation that would prevent this that I don't understand, but would it be possible to have the same concept in the views package with views.ConfigurationQueryParameters here without typeId and views.OutputConfigurationParameters that extends it to add typeId?

Reply:
I updated it as you suggested, plus changed the swagger doc generation to  use `allOf` schema tag for inheritance. It seems to work and it's cleaner.